awx/main/tasks.py

```diff
@@ -412,8 +412,7 @@ def _gather_and_ship(subset, since, until):
                 until = None
                 while start < gather_time:
                     until = start + timedelta(hours = 4)
-                    if (until > gather_time):
-                        until = gather_time
+                    until = min(until, gather_time)
```

Comment:
This will be made obsolete by my #9358 PR, and if this lands beforehand will make my life unnecessarily annoying.

Reply:
Okay I'll revert this. :sparkles: